### PR TITLE
feat: Use buildjet for ci-backend

### PR DESCRIPTION
**Pull request by ellie**
## Problem

We have fairly limited CI concurrency with GitHub, and also limited choice in runners

With Buildjet, we can get as much concurrency as we would like, and far more choice in runners. We also don't need to self host runners (risky - we're open source)

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: .github/workflows/ci-backend.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# This workflow runs all of our backend django tests.
+# This workflow runs all of our backend django tests..ci-back
```
Comment (pauldambra) [inline .github/workflows/ci-backend.yml:1]:
```suggestion
# This workflow runs all of our backend django tests.
```

typo?

**Review by pauldambra — APPROVED** [review]
Definitely worth trying...

I guess we need to "measure" speed of the tasks (since one of the selling points is speed)

And then we'd need to assess whether we want more concurrency (you can at least pay incrementally for concurrency with build jet instead of GitHub's huge bump to Enterprise)

**Comment by pauldambra** [comment]
Things still failing but I guess it's a work in progress. Definitely worth trying!

**Comment by ellie** [comment]
> Things still failing but I guess it's a work in progress. Definitely worth trying!

Yeah I was seeing if ARM was any faster, but

>  Error: The version '3.10.10' with architecture 'arm64' was not found for Ubuntu 22.04.

so I'll leave it as AMD64 for now

File: .github/workflows/ci-backend.yml (hunk 1 above)
Comment (ellie) [reply .github/workflows/ci-backend.yml:1]: oops, indeed

**Pull request merged**